### PR TITLE
feat(typography): refactor text styles

### PR DESCRIPTION
**Pull request by francisrupert**
# Refactor text styles

its-called-style-people-eugene-levy

## :hammer_and_wrench: Type Of Change

- [ ] Fix
- [x] Feature
- [ ] Performance Improvement
- [ ] Refactor

## :book: Jira Ticket

* [DLT-1610: Refactor Design Tokens](https://dialpad.atlassian.net/browse/DLT-1610)
* [DLT-1611: Implement refactored Design Tokens in Dialtone CSS](https://dialpad.atlassian.net/browse/DLT-1611)

## :book: Description

* Updating naming convention of text style classes, while still supporting previous ones (temporarily at least until they've been replaced in product).
* Added a few new variants
  * In `dialtone-tokens` via Tokens Studio. Previewable in [this Figma branch `text-styles-refactor`](https://www.figma.com/file/8RbAab7w2rQeg0Ii8jtaAl/branch/Thb9IBp18ppQYn70tYhCzy/DT9-Design-Tokens%3A-Text-%26-Color-Styles?type=design&node-id=3746-13426&mode=design&t=4wkBdx0pPbrpZVe8-11).
  * In `dialtone-css`
  * In `dialtone-documentation/docs/_data/type.json`
  * Applied updates in `dialtone-vue2` and `dialtone-vue3`.
  * Generated Type Styles in Figma Library branch (above Figma link, not part of this PR).

* Added a bit more doc'ing on Design Language > Typography

## :pencil: Checklist

For all PRs:

- [x] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [ ] I have reviewed my changes.
- [x] I have added all relevant documentation.
- [x] I have considered the performance impact of my change.

For all Vue changes:

- [ ] I have added / updated unit tests.
- [x] I have made my changes in Vue 2 and Vue 3. Note: you may sync your changes from Vue 2 to Vue 3 (or vice versa) using the `./scripts/dialtone-vue-sync.sh` script.
- [ ] I have validated components with a screen reader.
- [ ] I have validated components keyboard navigation.

For all CSS changes:

- [x] I have used design tokens whenever possible.
- [x] I have considered how this change will behave on different screen sizes.
- [x] I have visually validated my change in light and dark mode.
- [x] I have used gap or flexbox properties for layout instead of margin whenever possible.

## :crystal_ball: Next Steps

[DLT-1613: Implement refactored text styles in product codebase](https://dialpad.atlassian.net/browse/DLT-1613)

### CODE REVIEW

**Comment by francisrupert** [comment]
This may seem like a large PR, but that's only because of a flurry of search-and-replace, and the `dialtone-tokens` package's updates.

The primary page to see updates is https://dialtone.dialpad.com/deploy-previews/pr-198/design/typography/#usage

The main code updates to review are called out below...

File: apps/dialtone-documentation/docs/_data/type.json
Comment (francisrupert) [inline apps/dialtone-documentation/docs/_data/type.json:1]: Updated the type doc json to reflect new names and updated values.

File: apps/dialtone-documentation/docs/design/typography/index.md
Comment (francisrupert) [inline apps/dialtone-documentation/docs/design/typography/index.md:1]: Updated the Design Language > Typography page to reflect the refactored Text Style classes and provide a visualization to describe the naming convention

File: packages/dialtone-css/lib/build/less/utilities/typography.less
Comment (francisrupert) [inline packages/dialtone-css/lib/build/less/utilities/typography.less:1]: This `dialtone-css/lib/build/less/utilities/typography.less` is where the key style changes are made. Note there some comments above previous names being retained for backwards-compatibility, to be resolved later via https://dialpad.atlassian.net/browse/DLT-1614

File: packages/dialtone-tokens/tokens/$themes.json
Comment (francisrupert) [inline packages/dialtone-tokens/tokens/$themes.json:1]: This is impossible to PR review, but is effectively being reviewed via Figma's equivalent Merge Preview process with Francis, Fede, Yorbi.

File: packages/dialtone-tokens/tokens/base/default.json
Comment (francisrupert) [inline packages/dialtone-tokens/tokens/base/default.json:1]: This is impossible to PR review, but is effectively being reviewed via Figma's equivalent Merge Preview process with Francis, Fede, Yorbi.

File: packages/dialtone-tokens/tokens/semantic/dp/default.json
Comment (francisrupert) [inline packages/dialtone-tokens/tokens/semantic/dp/default.json:1]: This is impossible to PR review, but is effectively being reviewed via Figma's equivalent Merge Preview process with Francis, Fede, Yorbi.

**Review by yorbi-dp — APPROVED** [review]
This looks good to me! It's a significant improvement in how we handle types and more consistent with the rest of our naming conventions! 👍

**Review by braddialpad — APPROVED** [review]
Looks good but I have some concerns about risk. Can we be sure this isn't going to break a ton of stuff when updated in product? I see you have made backwards compatibility changes, but are the values going to be different for those existing classes than they were before?

**Comment by francisrupert** [comment]
@braddialpad 

> ...are the values going to be different for those existing classes than they were before

The raw values are the same between them by way of the associated CSS Variable. For example...

```
  line-height: var(--dt-typography-headline-sm-line-height);
  font-size: var(--dt-typography-headline-sm-font-size);
  font-weight: var(--dt-typography-headline-sm-font-weight);
```

replaces

```
  line-height: var(--dt-font-line-height-300);
  font-size: var(--dt-font-size-100);
  font-weight: var(--dt-font-weight-bold);
```

And the Design Tokens reference the originating value, e.g.
```
var(--dt-typography-headline-sm-line-height) == var(--dt-font-line-height-300)
```

Though that can only be seen in the `tokens.json`.

**Comment by braddialpad** [comment]
Alright we should be good then

File: packages/dialtone-vue2/recipes/conversation_view/emoji_row/emoji_row.vue
Comment (ninamarina) [inline packages/dialtone-vue2/recipes/conversation_view/emoji_row/emoji_row.vue]: Is there a ticket for fixing the style dictionary?

**Comment by francisrupert** [comment]
@braddialpad,

> ...when updated in product...

My understanding is that a Dialtone engineer will directly update the `dialtone` dependency, right? So when this specific version number is applied, it would be prudent to ensure that they validate.

File: packages/dialtone-vue2/recipes/conversation_view/emoji_row/emoji_row.vue
Comment (francisrupert) [reply packages/dialtone-vue2/recipes/conversation_view/emoji_row/emoji_row.vue]: Yeah: https://dialpad.atlassian.net/browse/DLT-1612

I was going to call it out to Brad and Julio since they're done most of the Style Dictionary stuff. I wasn't sure if it were straight-forward or needs a little care. I tried to figure it out, but probably shouldn't dare. Whatchathink @braddialpad?

It doesn't block this task, though would be lovely to resolve soon.

Comment (braddialpad) [reply packages/dialtone-vue2/recipes/conversation_view/emoji_row/emoji_row.vue]: I think style dictionary wise it should be fine, but I might have to make some updates to the postcss that generates the composition tokens. Do we have a mapping from old -> new?

```
--dt-typography-body-small = --dt-typography-body-sm
```

and so on..

Comment (francisrupert) [reply packages/dialtone-vue2/recipes/conversation_view/emoji_row/emoji_row.vue]: I'll add that mapping to https://dialpad.atlassian.net/jira/software/c/projects/DLT/issues/DLT-1612. Would you prefer to piggy-pack that change onto this PR, or its own task after this one merges?

Comment (braddialpad) [reply packages/dialtone-vue2/recipes/conversation_view/emoji_row/emoji_row.vue]: Yeah I'll just tack it on here, since we probably shouldn't merge this without it

Comment (francisrupert) [reply packages/dialtone-vue2/recipes/conversation_view/emoji_row/emoji_row.vue]: Sweet. I've added the mapping in the description field of https://dialpad.atlassian.net/browse/DLT-1612.

Additionally, I'm assuming this is the same culprit for why it doesn't appear [on the Tokens table](https://dialtone.dialpad.com/deploy-previews/pr-198/tokens/#typography). Just like for the CSS Variable, the typography property is available (e.g. `var(--dt-typography-body-md-font-family)`) but not the overall composition one, (e.g. `var(--dt-typography-body-md)`.

The best way to test your PostCSS update is the remove all the lines that have 
"`// remove once Style Dictionary transform resolved`".

**Comment by braddialpad** [comment]
Pushed my fix.. but appears github is broken right now..

Screenshot 2024-03-12 at 6 45 23 PM

Please validate it looks okay if it starts working again.

This will be a breaking change due to renaming of tokens. The utility classes won't break, but anything using the tokens directly will need to change, including android. We'll have to put out a communication about this and publish the mapping table on there.

**Comment by francisrupert** [comment]
> Please validate it looks okay if it starts working again.

Awesome! Thanks for reconciling. I'll give it a look and ensure all is good.

> ...We'll have to put out a communication about this and publish the mapping table on there.

I've logged [DLT-1624](https://dialpad.atlassian.net/browse/DLT-1624) which I'll have Yorbi do. Feel free to give that a look and add any detail you feel the description is missing.

[DLT-1624]: https://dialpad.atlassian.net/browse/DLT-1624?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by braddialpad** [comment]
Looks like github is back to normal and the latest preview has the changes

File: packages/dialtone-css/lib/build/less/utilities/typography.less
Comment (ninamarina) [inline packages/dialtone-css/lib/build/less/utilities/typography.less]:
```suggestion
.d-label-plain {
```

File: packages/dialtone-vue3/recipes/conversation_view/emoji_row/emoji_row.vue
Comment (ninamarina) [inline packages/dialtone-vue3/recipes/conversation_view/emoji_row/emoji_row.vue]: I think this is part of what needs to be replaced with the last changes from @braddialpad 
The same with the other cases that have `// replace with this once Style Dictionary transform resolved`

File: packages/dialtone-css/lib/build/less/utilities/typography.less
Comment (francisrupert) [reply packages/dialtone-css/lib/build/less/utilities/typography.less]: Ooh, good catch.

File: packages/dialtone-vue3/recipes/conversation_view/emoji_row/emoji_row.vue
Comment (francisrupert) [reply packages/dialtone-vue3/recipes/conversation_view/emoji_row/emoji_row.vue]: I got it.

**Pull request merged**
